File: testes_simples/códigos_quebrados/golang_quebrados/quebrado2.go
// Quebrado ainda
package main

import (
    "errors"
    "fmt"
)

type Usuario struct {
    Nome  string
    Idade int
}

type GerenciadorUsuarios struct {
    usuarios []Usuario
}

func (g *GerenciadorUsuarios) AdicionarUsuario(u Usuario) {
    g.usuarios = append(g.usuarios, u)
}

func (g *GerenciadorUsuarios) RemoverUsuario(nome string) error {
    for i, usuario := range g.usuarios {
        if usuario.Nome == nome {
            g.usuarios = append(g.usuarios[:i], g.usuarios[i+1:]...)
            return nil
        }
    }
    return errors.New("Usuário não encontrado")
}

func (g *GerenciadorUsuarios) MediaIdades() float64 {
    soma := 0
    for _, usuario := range g.usuarios {
        soma += usuario.Idade
    }
    return float64(soma) / float64(len(g.usuarios))
}

File: testes_simples/códigos_quebrados/python_quebrados/quebrado2.py
"""Quebrado ainda"""
class GerenciadorUsuarios:
    def __init__(self):
        self.usuarios = []
    
    def adicionar_usuario(self, nome, idade):
        self.usuarios.append({'nome': nome, 'idade': idade})
    
    def remover_usuario(self, nome):
        for usuario in self.usuarios:
            if usuario['nome'] == nome:
                self.usuarios.remove(usuario)
    
    def usuario_mais_velho(self):
        mais_velho = None
        for usuario in self.usuarios:
            if mais_velho is None or usuario['idade'] > mais_velho['idade']:
                mais_velho = usuario
        return mais_velho['nome']
    
    def media_idades(self):
        total = 0
        for usuario in self.usuarios:
            total += usuario['idade']
        return total / len(self.usuarios)